build_functions/CMakeLists.txt

```diff
@@ -193,3 +193,22 @@ function(compileTargetAsC11 theTarget)
     set_target_properties(${theTarget} PROPERTIES CXX_STANDARD 11)
   endif()
 endfunction()
+
+set(LIBS_TO_BE_REAL_CHECKED CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+
+function (add_library())
+    _add_library(${ARGV})
+    get_target_property(lib_type ${ARGV0} TYPE)
+    if( (NOT ${CMAKE_CURRENT_LIST_DIR} MATCHES "deps") AND # ignore dependency libs
+        (NOT ${ARGV0} MATCHES "_ut_lib|int_lib$") AND # ignore test libs
+        (NOT ${lib_type} STREQUAL "INTERFACE_LIBRARY")) # ignore interface libs
+        set(LIBS_TO_BE_REAL_CHECKED "${LIBS_TO_BE_REAL_CHECKED};${ARGV0}" CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+    endif()
+endfunction()
+
+function (get_target_files())
+    list(SUBLIST ARGV 1 -1 libraries)
+    foreach(lib ${libraries})
+        set(${ARGV0} "${${ARGV0}} \$\<TARGET_FILE:${lib}\>" CACHE INTERNAL "")
```

Comment:
>CACHE INTERNAL [](http://example.com/codeflow?start=60&length=14)

There's some funny mechanics with CACHE INTERNAL. Because once it is set, it might remain set between different runs of cmake. And that might not be wanted when a developer is adding let's say a new target to the project.

So I am not sure what's the intention here. Will it all be revealed when the custom target gets added?

I think the review needs the custom target before it can be checked in. Reason to wait.

Reply:
I clear the variable in every run.